pkg/app/execontext.go

```diff
@@ -167,7 +190,16 @@ func (ref *Output) State(state string, params ...OutVars) {
 	}
 	defer color.Unset()
 
-	fmt.Printf("cmd=%s state=%s%s%s%s\n", ref.CmdName, state, exitInfo, sep, info)
+	//marshal info to json
+	if ref.JSONFlag == "json" {
+		var jsonData []byte
```

Comment:
This is similar to the other calls, but a bit different because we have the `params` slice (of maps where each map is a set of key/values), which we need to convert into the individual json fields. There needs to be a json output equivalent to the logic that creates the `info` string above.

Reply:
Great suggestion, have tried to fix this, do check, thanks!